src/Servers/Kestrel/Core/src/Internal/Infrastructure/IMultiplexedTransportManager.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System.Net;
+using Microsoft.AspNetCore.Connections;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure;
+
+internal interface IMultiplexedTransportManager
+{
+    bool HasFactories { get; }
+
+    Task<EndPoint> BindAsync(EndPoint endPoint, MultiplexedConnectionDelegate multiplexedConnectionDelegate, ListenOptions listenOptions, CancellationToken cancellationToken);
+
+    Task StopAsync(CancellationToken cancellationToken);
+    Task StopEndpointsAsync(List<EndpointConfig> endpointsToStop, CancellationToken cancellationToken);
+}
```

Comment:
Instead of adding `IMultiplexedTransportManager` and `ITransportManager`, could we register `IMultiplexedHttpsFeatureFactory`? I know it's internal, but it seems like the smaller change unless there's some other benefit to splitting things up.

```
internal interface IMultiplexedHttpsFeatureFactory
{
    void PopulateHttpsFeatures(IFeatureCollection features, ListenOptions listenOptions);
}
```

We could have the default implementation throw an `InvalidOperationException` similar to `InvalidUseHttpsHelper`.

Reply:
Sorry, I'm not sure I understand this suggestion.  (I had assumed I could go study `IMultiplexedHttpsFeatureFactory`, but I think you just invented it?)  I think you might be suggesting that we add the pay-for-play functionality to a (the?) feature collection, rather than the DI container?

Reply:
Implemented in #47454.